### DISCUSSION TITLE
problem with commissioning end devices

### DISCUSSION THREAD

**Discussion by sohaibbinaltaf** [openthread/openthread#8412] [Q&A] [answered]
I have developed a thread network with three devices, raspberry pi with nRF52 dongles. I can ping each node and the network is working fine. All the three nodes are REED (router enabled end devices) with ot-rcp.hex file flashed in the dongles. These devices can turn into routers, but i want to have a device which is dedicatedly an end device. I have a few questions:

1) Do I need to have a ot-cli-ftd.hex file for this end device? I can see in codelabs that ot-cli-ftd.hex is also used as commissioner (which i guess also works as a router). 

2) I am trying in-band commissioning process, but my joiner returns this answer after the following command:

joiner start J01NMEME
Error 13: InvalidState

is this because of the rcp.hex file or some other reason?

3) I am trying to do the same using ot-ftd-cli.hex, but it doesn't respond to the ot-ctl commands. Can you please comment if there is something missing?

$ sudo ot-ctl
**connect session failed: Connection refused**

$ dataset set active 0e080000000000010000000300000d35060004001fffe002081fde57d326d560770708fdbc68a740973ee50510c047914315e34bc9848c27c8e4874e75030f4f70656e5468726561642d653464380102e4d804108aad4605fef84ece5b3e97c6b6cae0990c0402a0f7f8
**bash: dataset: command not found**

4) What would be the difference between rcp.hex and  ot-ftd-cli.hex files if both of these can be used to build a router ? 

Most importantly, as i want a device (raspberry pi with nRF52 dongle) which can only stay as end device and not function or change into a router, is it possible to build it with the rcp.hex  file .

**Comment by bukepo** [comment, marked as answer]
1. You may try disabling router eligible of the dedicatedly end device:

```bash
routereligible disable
```
2. Try bring up the interface before issuing `joiner start`:
```bash
ifconfig up
```
3. If you are using raspberry pi, you should stick to ot-rcp.hex as the firmware of nRF52. Please check the platform designs: https://openthread.io/platforms#platform_designs
4. You may check the platform designs: https://openthread.io/platforms#platform_designs. There are discussions of single-chip and co-processor designs.

**Reply by sohaibbinaltaf** [reply]
Thanks a lot for your detailed response. Let me check it.

**Comment by sohaibbinaltaf** [comment]
Hello, 

After the disable router functionality command, the router remains as child node, but I have observed it switches from one router to another. How can we keep this child node connected with one and only one node. I have tried the below commands to connect it to a certain router but still switches to another router  after some time:

On router 
$ sudo ot-ctl dataset active -x
0e080000000000010000000300001235060004001fffe002083d3818dc1c8db63f0708fda85ce9df1e662005101d81689e4c0a32f3b4aa112994d29692030f4f70656e5468726561642d35326532010252e204103f23f6b8875d4b05541eeb4f9718d2f40c0302a0ff
Done

On child node

$ sudo ot-ctl dataset set active  (I added sudo ot-ctl)
0e080000000000010000000300001235060004001fffe002083d3818dc1c8db63f0708fda85ce9df1e662005101d81689e4c0a32f3b4aa112994d29692030f4f70656e5468726561642d35326532010252e204103f23f6b8875d4b05541eeb4f9718d2f40c0302a0ff
Done

**Reply by jwhui** [reply]
Because wireless connectivity changes over time, Thread intentionally adapts to changing conditions.

If you really want to restrict what nodes a given Thread device can communicate with, take a look at the [`macfilter`](https://github.com/openthread/openthread/blob/main/src/cli/README.md#macfilter) feature. Note, however, that you should only use this feature for testing.

**Reply by sohaibbinaltaf** [reply]
Thank you @jwhui for your response.
